tests/functions/plot_test.py

```diff
@@ -222,3 +222,33 @@ def test_raises_ValueError_when_given_binned_data():
     da = sc.data.table_xyz(100).bin(x=10, y=20)
     with pytest.raises(ValueError, match='Cannot plot binned data'):
         pp.plot(da)
+
+
+def test_use_non_dimension_coords_strings():
+    da = data_array(ndim=2, binedges=True)
+    da.coords['xx2'] = 7.5 * da.coords['xx']
+    da.coords['yy2'] = 3.3 * da.coords['yy']
+    p = pp.plot(da, coords=['xx2', 'yy2'])
+    assert p.dims['x'] == 'xx2'
+    assert p.dims['y'] == 'yy2'
+    assert p.canvas._xmax == 7.5 * da.coords['xx'].max().value
+    assert p.canvas._ymax == 3.3 * da.coords['yy'].max().value
+
+
+def test_use_non_dimension_coords_variables():
+    da = data_array(ndim=2, binedges=True)
+    xx2 = 7.5 * da.coords['xx']
+    yy2 = 3.3 * da.coords['yy']
+    p = pp.plot(da, coords=[xx2, yy2])
+    assert p.dims['x'] == 'xx'
+    assert p.dims['y'] == 'yy'
```

Comment:
This is really weird. Why should passing strings lead to dim renaming, but passing coords not?

Reply:
Sure, but why are you renaming dims in the other case?

Reply:
To get the name of the new coord along the x and y axes?

Reply:
I find this inconsistent, I would have expected
```python
pp.plot(da, coords=['xx2'])
pp.plot(da, coords=[da.coords['xx2']])
```
to give the same result. If not, it may indicate that the API design is not correct?